### PR TITLE
This PR will add the following changes if applied:

### PR DESCRIPTION
**Pull request by jaredmorgs**
- all aspects of algolia plug-in configuration using step-by-step instructions.
- made the book.json syntax match the current gitbook formatting.
- how to add an environment variable to a book through the editor.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,23 @@ This plugins requires gitbook `>=3.0.0`.
...
+
+### Set up the plugin in your book.
+
+1. Open Settings > Plugins Store interface in the Book Editor.
+2. Install the `algolia` plugin.
+3. Specify the following values in the modal for your book:
+    * **Index ID**, which is the name of the index you set up in Algolia web.
+    * **Application ID**, which is your unique application ID for API identification. 
+    * **Search-only API Key**, which is the unique ID for search queries.
+    * **Is your algoria account free**, which you leave as `true` if you have what Algoria call a "Hacker Account".
```
Comment (redox) [inline README.md]: `s/Algoria/Algolia/` small typo :)

Comment (jaredmorgs) [inline README.md]: Good catch @redox . I've corrected this.

File: README.md [hunk 2]
```diff
@@ -6,7 +6,23 @@ This plugins requires gitbook `>=3.0.0`.
...
+1. Sign up to [Algolia](https://www.algolia.com)
+2. Set up a basic index.
+3. Access your [API keys settings](https://www.algolia.com/api-keys), and leave this page open.
+
+### Set up the plugin in your book.
+
+1. Open Settings > Plugins Store interface in the Book Editor.
+2. Install the `algolia` plugin.
+3. Specify the following values in the modal for your book:
+    * **Index ID**, which is the name of the index you set up in Algolia web.
```
Comment (redox) [inline README.md]: I would go for `Index Name` instead of `Index ID`, but minor wording issue.

**Comment by Soreine** [comment]
Thank you :)

**Pull request merged**